src/librustdoc/core.rs

```diff
@@ -570,6 +607,68 @@ pub fn run_core(options: RustdocOptions) -> (clean::Crate, RenderInfo, RenderOpt
     })
 }
 
+thread_local!(static DEFAULT_TYPECK: for<'tcx> fn(TyCtxt<'tcx>, LocalDefId) -> &'tcx ty::TypeckTables<'tcx> = {
+    let mut providers = ty::query::Providers::default();
+    rustc_typeck::provide(&mut providers);
+    providers.typeck_tables_of
+});
```

Comment:
This would probably fit better in `local_providers.typeck_tables_of = move |tcx, def_id| {`.

Unless you want to create a `fn hooked_typeck_tables_of` function or similar, instead of using a closure for it.